runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7967,6 +7967,21 @@ J9::X86::TreeEvaluator::VMnewEvaluator(
 
       dataOffset = TR::Compiler->om.objectHeaderSizeInBytes(); //Not used...
       classReg = node->getFirstChild()->getRegister();
+
+      // For Non-SVM AOT, the class has to be in a register
+      if (TR::Compiler->om.areValueTypesEnabled() &&
+          !classReg &&
+          cg->needClassAndMethodPointerRelocations() &&
+          !comp->getOption(TR_UseSymbolValidationManager))
+         {
+         classReg = cg->evaluate(node->getFirstChild());
+
+         if (comp->getOption(TR_TraceCG))
+            {
+            traceMsg(comp, "%s: evaluate loadaddr: clazz %p classReg %s\n", __FUNCTION__, clazz, classReg ? classReg->getRegisterName(comp) : "");
```

Comment:
You might want to change the empty string to "\<none\>" or something like that to make the case where there is no `classReg` easier to notice.

Reply:
Addressed in a59bddb